### ISSUE TITLE
support json comparison in assertions

### ISSUE THREAD

**Issue by mathnogueira** [kubeshop/tracetest#3684]
## **Describe the enhancement you'd like to see**
Comparing all important data from a large JSON object can be cumbersome. In JS testing tools, they have an assertion function to check if an object is a subset of another, with this approach they gain two advantages: 
1. They can compare several information at once
2. They can ignore some fields

I suggest we implement something like that. We could even use the existing `contains` comparator to compare two JSON objects.

### Rules
1. If one single value defined on the right side of the statement is missing on the left side, the assertion fails;
2. Extra fields on the left side of the statement are ignored and don't affect the outcome of the assertion;
3. If one of the compared values is an array, all items must match, including the order **(Should we be more flexible?)**
4. It must be possible to deeply compare inner objects;

This would only be a JSON comparison if both sides of the statement are resolved as strings and can be parsed as JSON.

#### 1. Order of attributes don't matter
```
'{ "name": "john", "age": 32, "email": "john@company.com" }' contains '{ "name": "john", "email": "john@company.com" }'
```

#### 2. Inner objects can be partially compared as well
```
'{ "name": "john", "age": 32, "email": "john@company.com", "company": { "name": "company", "address": "123 Secord Street" } }' contains '{ "name": "john", "email": "john@company.com", "company": { "name": "company" } }'
```

**Additional context**
This was asked by one of our users in our Community Slack workspace.

**Comment by danielbdias** [comment]
It makes sense. On the inner objects case, I would add one more case: a `list` comparison:

```
'[{ "name": "john", "age": 32, "email": "john@company.com" }, { "name": "jane", "age": 30, "email": "jane@company.com" }]' 
contains 
'{ "name": "john", "email": "john@company.com" }'
```

or even like this:
```
'{ "organization": "Company", "employees": [{ "name": "john", "age": 32, "email": "john@company.com" }, { "name": "jane", "age": 30, "email": "jane@company.com" }] }' 
contains 
'{ "name": "john", "email": "john@company.com" }'

```

**Comment by mathnogueira** [comment]
I think the first option is easier to get a grasp of. Assertions already support arrays, so that would be an easier change to introduce

**Comment by jorgeepc** [comment]
Looks good. Just one comment, I think the `toMatchObject` from `jest` has a different behavior for the `list comparison`: The method returns true only if each object in the received array matches the expected array.

```js
expect( [{foo: 'bar'}, {baz: 1, extra: 'quux'}] ).toMatchObject([
  {foo: 'bar'},
  {baz: 1},
]);
```

**Comment by xoscar** [comment]
@jorgeepc I think it would be very interesting to research how jest is doing it

**Comment by mathnogueira** [comment]
From [jest](https://jestjs.io/docs/expect#tomatchobjectobject)

> You can also pass an array of objects, in which case the method will return true only if each object in the received array matches (in the toMatchObject sense described above) the corresponding object in the expected array. This is useful if you want to check that two arrays match in their number of elements, as opposed to arrayContaining, which allows for extra elements in the received array.

This means that both sides would have to be arrays for that to work.

**Comment by jorgeepc** [comment]
This was done and released in [v0.16.0](https://github.com/kubeshop/tracetest/releases/tag/v0.16.0) Closing the ticket
